src/util/psbt/serialize.rs

```diff
@@ -144,3 +155,172 @@ impl Deserialize for SigHashType {
         }
     }
 }
+
+// Taproot related ser/deser
+impl Serialize for schnorr::PublicKey {
+    fn serialize(&self) -> Vec<u8> {
+        self.serialize().to_vec()
+    }
+}
+
+impl Deserialize for schnorr::PublicKey {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        schnorr::PublicKey::from_slice(bytes)
+            .map_err(|_| encode::Error::ParseFailed("Invalid xonly public key"))
+    }
+}
+
+impl Serialize for (schnorrsig::Signature, SchnorrSigHashType)  {
+    fn serialize(&self) -> Vec<u8> {
+        let mut ser_sig = self.0.as_ref().to_vec();
+        if let SchnorrSigHashType::Default = self.1 {
+            // No need to append for default type
+        } else {
+            ser_sig.push(self.1  as u8);
+        }
+        ser_sig
+    }
+}
+
+impl Deserialize for (schnorrsig::Signature, SchnorrSigHashType) {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        match bytes.len() {
+            65 => {
+                let hash_ty = SchnorrSigHashType::from_u8(bytes[64])
+                    .map_err(|_| encode::Error::ParseFailed("Invalid Sighash type"))?;
+                let sig = schnorrsig::Signature::from_slice(&bytes[..64])
+                    .map_err(|_| encode::Error::ParseFailed("Invalid Schnorr signature"))?;
+                Ok((sig, hash_ty))
+            }
+            64 => {
+                let sig = schnorrsig::Signature::from_slice(&bytes[..64])
+                    .map_err(|_| encode::Error::ParseFailed("Invalid Schnorr signature"))?;
+                Ok((sig, SchnorrSigHashType::Default))
+            }
+            _ => Err(encode::Error::ParseFailed("Invalid Schnorr signature len"))
+        }
+    }
+}
+
+impl Serialize for (schnorr::PublicKey, TapLeafHash) {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf = self.0.serialize().to_vec();
+        buf.extend(self.1.as_ref());
+        buf
+    }
+}
+
+impl Deserialize for (schnorr::PublicKey, TapLeafHash) {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        if bytes.len() < 32 {
+            return Err(io::Error::from(io::ErrorKind::UnexpectedEof).into())
+        }
+        let a: schnorr::PublicKey = Deserialize::deserialize(&bytes[..32])?;
+        let b: TapLeafHash = Deserialize::deserialize(&bytes[32..])?;
+        Ok((a, b))
+    }
+}
+
+impl Serialize for ControlBlock {
+    fn serialize(&self) -> Vec<u8> {
+        self.serialize()
+    }
+}
+
+impl Deserialize for ControlBlock {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        Self::from_slice(bytes)
+            .map_err(|_| encode::Error::ParseFailed("invalid control block"))
```

Comment:
Throwing out inner error is not great.

Reply:
Unfortunately, the existing code also threw away errors so I cut some corners here. The trait return error (`encode::Error`) type is also unrelated to `pset::serialize::Deserialize`. Cleaning up this might require touching other parts of the code that are not taproot psbt related which I why I skipped it.